### PR TITLE
Fix Log family table return wrong rows count after truncate

### LINKED ISSUE #50578: LOG engine count() after TRUNCATE is not empty
> A clear and concise description of what works not as it is supposed to.

When I do TRUNCATE on LOG ENGINE table, it deletes records, but count() is not empty

> A link to reproducer

https://fiddle.clickhouse.com/9ee5ac13-56b9-4b78-9f61-66f0e8e49de6

**Does it reproduce on recent release?**
Yes

**How to reproduce**

* Which ClickHouse server version to use
23.4.2.11
* Which interface to use, if matters
fiddle
* `CREATE TABLE` statements for all tables involved
```
CREATE TABLE default.test_log
(
    `crypto_name` String,
    `trade_date` Date
)
ENGINE = Log
SETTINGS index_granularity = 8192;
```

* Sample data for all these tables
`INSERT INTO default.test_log (crypto_name, trade_date) VALUES ('abc', '2021-01-01'), ('def', '2022-02-02');`
* Queries to run that lead to unexpected result
```
TRUNCATE TABLE default.test_log;
SELECT count(*) FROM  default.test_log;
```
2
**Expected behavior**
count is 0
> A clear and concise description of what you expected to happen.

as `select` from the table after `truncate` returns nothing, `count` should return 0

### PR DESCRIPTION
**Pull request by ucasfl**
### Changelog category (leave one):

- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix Log family table return wrong rows count after truncate. Closes #50578.

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit f5b3e346b7274046d4d724265f7f7800dfb8d976 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50585/f5b3e346b7274046d4d724265f7f7800dfb8d976/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
Stateless tests flaky check (asan) - the new test has failed

File: tests/queries/0_stateless/02771_log_faminy_truncate_count.sql [hunk 1]
```diff
@@ -1,3 +1,4 @@
+-- Tags: no-parallel
 DROP TABLE IF EXISTS default.test_log;
 CREATE TABLE default.test_log
```
Comment (tavplubix) [inline tests/queries/0_stateless/02771_log_faminy_truncate_count.sql:3]: Can we just remove both `default` and `no-parallel`? `clickhouse-test` runs each test in a separate database to allow parallel execution

**Comment by tavplubix** [comment]
Integration tests (asan) [6/6] - #49400

**Pull request merged**